### PR TITLE
Add Python 3 support

### PR DESCRIPTION
**Pull request by lwgordonimos**

### CODE REVIEW

File: aodndata/soop/soop_co2_nrt_nc_generator.py [hunk 1]
```diff
@@ -291,7 +292,7 @@ def check_parameters(dataf, vessel_code, input_param, src_file):
         # var TYPE conversion to string outside this function
         for param in rt_input_parameters:
             if param not in set(['Type', 'PcDate', 'PcTime']):
-                dataf[param] = dataf[param].apply(pd.to_numeric, errors=coerce)
+                dataf[param] = dataf[param].apply(pd.to_numeric, errors='coerce')
```
Comment (lwgordonimos) [inline aodndata/soop/soop_co2_nrt_nc_generator.py]: @bpasquer as discussed, this coerce function 1) appears that it should be a string and 2) breaks on Python 3

Comment (lwgordonimos) [reply aodndata/soop/soop_co2_nrt_nc_generator.py]: @lbesnard are you able to have a look at this? 

It looks like the intention was to supply the string 'coerce' to the to_numeric method, but it had a reference to the coerce *function* supplied. 

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.to_numeric.html

Updating this to the string and running on Python 3, which has no builtin coerce function, the tests fail as follows:

```bash
======================================================================
FAIL: test_good_rt_aa_txt (test_aodndata.soop.test_soop_co2.TestSoopCo2Handler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lwgordon/RubymineProjects/chef/src/python-aodndata/test_aodndata/soop/test_soop_co2.py", line 105, in test_good_rt_aa_txt
    check_params={'checks': ['cf', 'imos:1.4']})
  File "/home/lwgordon/RubymineProjects/chef/src/python-aodndata/.tox/py3/lib/python3.6/site-packages/aodncore/testlib/handlertest.py", line 39, in run_handler
    self.assertIsNone(handler.error)
AssertionError: ValueError("Variable 'TYPE' has dimensions ['TIME', 'string_10'] and shape (1087, 40), inconsistent with dimension sizes defined in template (1087, 10)",) is not None

======================================================================
FAIL: test_good_rt_in_txt (test_aodndata.soop.test_soop_co2.TestSoopCo2Handler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lwgordon/RubymineProjects/chef/src/python-aodndata/test_aodndata/soop/test_soop_co2.py", line 84, in test_good_rt_in_txt
    check_params={'checks': ['cf', 'imos:1.4']})
  File "/home/lwgordon/RubymineProjects/chef/src/python-aodndata/.tox/py3/lib/python3.6/site-packages/aodncore/testlib/handlertest.py", line 39, in run_handler
    self.assertIsNone(handler.error)
AssertionError: ValueError("Variable 'TYPE' has dimensions ['TIME', 'string_10'] and shape (958, 40), inconsistent with dimension sizes defined in template (958, 10)",) is not None
```

This is the only thing in this package failing on Python 3, so would be good to get it sorted out!

**Comment by lbesnard** [comment]
@bpasquer could you please have a look at my latest commit? I changed the co2 generator. It works fine with me, I pushed successfully a file on nec4

**Comment by lwgordonimos** [comment]
@mhidas @lbesnard @bpasquer are you happy for us to now merge this, so we can start testing on 9-nec-hob?

**Comment by lbesnard** [comment]
I'm happy. Are you happy?

**Comment by mhidas** [comment]
Can we use the new [pipeline integration testing](https://github.com/aodn/utilities/tree/master/pipeline/talend_test) utility to test these in the po box?

**Comment by mhidas** [comment]
As far as I understand, that would allow us to run a full suite (well, what's been configured so far) of pipeline/harvester tests. This is something we can't currently do on 9-nec-hob.

**Comment by lbesnard** [comment]
regression testing was performed successfully for the most part. Only issue is regarding the SOOP_CO2 harvester part.

Pushing a file (/mnt/archive/IMOS/SOOP/SOOP-CO2/VNAA_Aurora-Australis/REALTIME/2019/2/AA_2019-046-0001dat.txt from 10-nec-hob), the harvester complains about the variable TYPE being longer than character varying (3). And indeed the variable is up to 10 characters. @bpasquer what do you think? probably should update the harvester

**Comment by lbesnard** [comment]
disregard last comment. Was due to old talend databag in chef repo leading  SOOP CO2 RT files to the wrong harvester

**Pull request merged**